### PR TITLE
Add RFID to user

### PR DESCRIPTION
**Pull request by vgonda**
A new install will be creating pours using RFID. We are adding RFID to the user model, using it if provided to create a pour, and including it in the json list of users.

[[Contributes #147199951]](https://www.pivotaltracker.com/story/show/147199951)

### CODE REVIEW

File: app/controllers/api/v1/stats_controller.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@ module Api
   module V1
     class StatsController < ApiController
       def index
-        user = User.find params[:user_id]
+        user = User.where("id = ? OR rfid = ?", params[:user_id], params[:user_rfid]).first
```
Comment (albus522) [inline app/controllers/api/v1/stats_controller.rb]: To maintain the same functionality this needs to use `find_by!` to trigger a RecordNotFound error

File: app/controllers/pours_controller.rb [hunk 2]
```diff
@@ -13,7 +13,7 @@ def new
 
   def create
     tap = BeerTap.find(params[:beer_tap_id])
-    user = User.find(params[:user_id]) if params[:user_id] != '0'
+    user = User.where("id = ? OR rfid = ?", params[:user_id], params[:user_rfid]).first if params[:user_id] != '0'
```
Comment (albus522) [inline app/controllers/pours_controller.rb]: Same `find_by!` here

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -8,7 +8,7 @@ def index
   end
 
   def show
-    @user = User.find(params[:id])
+    @user = User.where("id = ? OR rfid = ?", params[:user_id], params[:user_rfid]).first
```
Comment (albus522) [inline app/controllers/users_controller.rb]: This uses `params[:id]` not `params[:user_id]`. Also needs to use `find_by!`

**Comment by vgonda** [comment]
Thanks for the review, David. I opted to go with `.first!` over `.find_by!`, as `find_by!` uses `AND`, and we want `OR` in this case. Using `.first!` will give us the `RecordNotFound` if a user does not exist with that user id or RFID.

**Pull request merged**